src/Serializers/QueryResultSerializer.php

```diff
@@ -81,16 +81,26 @@ public static function getSerialization( DataItem $dataItem, $printRequest = nul
 			case DataItem::TYPE_WIKIPAGE:
 				// Support for a deserializable _rec type with 0.6
 				if ( $printRequest !== null && strpos( $printRequest->getTypeID(), '_rec' ) !== false ) {
+					$diProperty = $printRequest->getData()->getDataItem();
+
+					if ( $printRequest->isMode( \SMW\Query\PrintRequest::PRINT_CHAIN ) ) {
+						$diProperty = $printRequest->getData()->getLastPropertyChainValue()->getDataItem();
+					}
```

Comment:
Isn't there a method on $printRequest that does this? If not, the new code is better added as a new method on that class